packages/front-end/components/Settings/ExperimentCheckListModal.tsx

```diff
@@ -0,0 +1,104 @@
+import {
+  ChecklistTask,
+  ExperimentLaunchChecklistInterface,
+} from "back-end/types/experimentLaunchChecklist";
+import { useState } from "react";
+import { FaPlus } from "react-icons/fa";
+import { useAuth } from "@/services/auth";
+import Modal from "../Modal";
+import SortableExperimentChecklist from "./SortableExperimentChecklist";
+import ExperimentChecklistEmptyState from "./ExperimentChecklistEmptyState";
+import NewExperimentChecklistItem from "./NewExperimentChecklistItem";
+
+export default function ExperimentCheckListModal2({
+  close,
+  mutate,
+  checklist,
+}: {
+  close: () => void;
+  mutate: () => void;
+  checklist?: ExperimentLaunchChecklistInterface;
+}) {
+  const { apiCall } = useAuth();
+  const [experimentLaunchChecklist, setExperimentLaunchChecklist] = useState<
+    ChecklistTask[]
+  >(checklist?.tasks ? checklist.tasks : []);
+  const [newTaskInput, setNewTaskInput] = useState<ChecklistTask | undefined>(
+    undefined
+  );
+
+  async function handleSubmit() {
+    const tasks = experimentLaunchChecklist;
+
+    if (tasks.length && tasks[tasks.length - 1].task === "") {
+      tasks.pop();
+    }
```

Comment:
nit pick suggestion; would this suffice?
```suggestion
    const tasks = experimentLaunchChecklist.filter(t => !!t.task);
```